src/hooks/useArticleTextSize.js

```diff
@@ -2,12 +2,14 @@ import { useState, useLayoutEffect } from 'preact/hooks'
 import { useSoftkey } from 'hooks'
 import { articleTextSize } from 'utils'
 
-export const useArticleTextSize = (origin, dependencies = []) => {

```

Comment:
I really don't like the idea of having `withSoftkeys` param here, it used to be apply this softkey everywhere but it's now allowed in `TextSize` screen only.

**Quick idea :**

1) the `useArticleTextSize` in most cases are doing only `articleTextSize.init()` only
2) move the `useSoftkey` only in `TextSize` component?

**Another broader idea :**

1) Since using `origin` a lot in every place, and this string becomes important how we maintain the softkey, let's maintain the `origin` as the object which holds all the origin string value. {}
2) when having this object, within`useArticleTextSize` we only enable the softkey in `origin.textSize` << means textSize screen
3) OR combined the idea above.

Reply:
Yea actually looking back at this idea, I dislike it too, not sure what I was thinking. How about https://github.com/wikimedia/wikipedia-kaios/pull/290/commits/6f9597da50c0af5765b8b564572e9a87dbbee14f? Indeed we can simplify a bit by having the hook only `init` components that wish to be 'subscribed' to the font size changes. And then abstract the actual softkey setting part only to `TextSize` component.